### PR TITLE
Modern CSS styleguide

### PR DESCRIPTION
**Pull request by aalin**

### CODE REVIEW

File: styleguide/css/MODERN-CSS.md [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+CSS Modules:
+
+``` css
+/* ItemListing.module.css */
+
+.itemListing { … }
+.inactive { … }
+.title { … }
+```
+``` jsx
```
Comment (olleolleolle) [inline styleguide/css/MODERN-CSS.md:70]: Minor: Will we need a newline between these backticks sections?

File: styleguide/css/MODERN-CSS.md [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+
+TODO: We still need to write a stylelint config, and make sure it's compatible with prettier so that it doesn't complain over rules that would be fixed by prettier.
+
+
+### Use [postcss](https://postcss.org/) with [postcss-preset-env](https://preset-env.cssdb.org/)
+
+[PostCSS Preset Env](https://preset-env.cssdb.org/) lets you convert modern CSS into something most browsers can understand, determining the polyfills you need based on your targeted browsers or runtime environments, using [cssdb](https://cssdb.org/).
+
+
+### Use CSS Modules
```
Comment (henrik) [inline styleguide/css/MODERN-CSS.md:33]: I'm not sure about this for reasons given in https://github.com/barsoom/auctionet/pull/147/files#r636957608.

Let's discuss it sometime after the deadline work we're doing now, when I have more mindspace. Possibly we want to wait until Fred joins us to make big decisions about changing how we do CSS?

File: styleguide/javascript/README.md [hunk 3]
```diff
@@ -82,22 +88,123 @@ Use arrows for functions that are passed as callbacks, since arrow functions bin
...
+
+Read and understand [the React documentation](https://reactjs.org/docs/hello-world.html). It's pretty short.
 
 ### Wrap your app in `<React.StrictMode>`
 
 This will enable some extra warnings in development mode.
 
 [React.js docs about Strict Mode](https://reactjs.org/docs/strict-mode.html)
 
+### Prefer function components over class components
```
Comment (henrik) [inline styleguide/javascript/README.md:127]: No opinion on this – happy for this part to be merged without the CSS stuff (for now).

File: styleguide/javascript/README.md [hunk 4]
```diff
@@ -82,22 +88,123 @@ Use arrows for functions that are passed as callbacks, since arrow functions bin
...
 If you're using function-style components in React, you (usually) don't need `this`, so it's better to use named functions because they are easier to read. But always prefer arrows instead of `.bind(this)`, and for unnamed closures.
 
+### camelCase in object properties
+
+Do:
+
+``` javascript
+const obj = {
+  fooBar: {
+    barBaz: 123
```
Comment (henrik) [inline styleguide/javascript/README.md:98]: Trailing commas

File: styleguide/javascript/README.md [hunk 5]
```diff
@@ -82,22 +88,123 @@ Use arrows for functions that are passed as callbacks, since arrow functions bin
 
 If you're using function-style components in React, you (usually) don't need `this`, so it's better to use named functions because they are easier to read. But always prefer arrows instead of `.bind(this)`, and for unnamed closures.
 
+### camelCase in object properties
```
Comment (henrik) [inline styleguide/javascript/README.md:91]: I'm not sure I agree with this. I seem to remember such tools making things a bit messier and less obvious than just keeping snake case. But I don't feel very strongly (I only remember it vaguely…) and I'm OK with us trying it out :)

File: styleguide/javascript/README.md [hunk 6]
```diff
@@ -82,22 +88,123 @@ Use arrows for functions that are passed as callbacks, since arrow functions bin
...
+
+Try to keep your components small and compose them instead of having large components that have too much state and logic.
+
 ### PropTypes
 
 Read [Typechecking With PropTypes](https://reactjs.org/docs/typechecking-with-proptypes.html).
 
 Our eslint rules will complain if you don't use them.
 
+### Use CSS Modules
```
Comment (henrik) [inline styleguide/javascript/README.md:200]: Again, let's discuss this part later.

File: styleguide/css/MODERN-CSS.md [hunk 7]
```diff
@@ -0,0 +1,138 @@
+![Barsoom](http://barsoom.se/barsoom.png)
```
Comment (henrik) [inline styleguide/css/MODERN-CSS.md:1]: We don't appear to be linking to this file anywhere.

I'm also not sure about having two CSS styleguides or using the term "modern" to distinguish them. Let's discuss this later per other comment.

File: styleguide/javascript/README.md (hunk 4 above)
Comment (aalin) [reply styleguide/javascript/README.md:98]: True

**Review by henrik — CHANGES_REQUESTED** [review]

File: styleguide/javascript/README.md (hunk 5 above)
Comment (aalin) [reply styleguide/javascript/README.md:91]: Works fine, just make sure you wrap all your API code in it and make it transparent.

With React it's also super-easy to inspect whatever data you're dealing with, so any issues you might have had with other libraries in the past might not even exist in React.

File: styleguide/css/MODERN-CSS.md (hunk 7 above)
Comment (aalin) [reply styleguide/css/MODERN-CSS.md:1]: We need separate styleguides for `app/assets` and `app/webpacker_assets`. `app/assets` should be considered legacy.

File: styleguide/javascript/README.md (hunk 5 above)
Comment (aalin) [reply styleguide/javascript/README.md:91]: camelCase is used for properties almost everywhere in the javascript world. Encountering snake_case all of a sudden just feels really weird...

Also helps with destructuring:

```
const { imageUrls } = item
// is more convenient than
const { image_urls: imageUrls } = item
// and this next one will make eslint angry
const { image_urls } = item

File: styleguide/css/MODERN-CSS.md (hunk 2 above)
Comment (aalin) [reply styleguide/css/MODERN-CSS.md:33]: CSS modules are basically the same as BEM, but automated.

File: styleguide/javascript/README.md (hunk 5 above)
Comment (henrik) [reply styleguide/javascript/README.md:91]: Good points. To be clear, I'm 👍  on this part :)

**Pull request closed without merging**